docs/solution-design.md

```diff
@@ -6,6 +6,50 @@ While there are many ways to achieve the goal we are looking for, for the purpos
 
 ## Components used and their roles
 
-## Message Flow
+The following components will be used as part of the solution:
+  
+- Cosmos DB Account
+- Cosmos DB Containers
+  - On Hand (onhand)
+  - Shipments (shipments)
+  - Master Data Container (mdc)
+- Azure Functions
+  - Change Feed Listeners
+  - Durable Function Orchestrator
+  - Durable Entity as a Virtual Actor
+
+As part of the challenges, we will deploy the Cosmos DB Account and Containers on Azure and build the Azure Functions to process the changes.
+
+## Processing Sequence
+
+The following diagram describes the interaction between the various components:
+
+![Processing Flow](./images/OpenHack_ProcessingFlow.png)
+
+The processing starts when documents are fed in to the Cosmos DB containers (shipments and on hand), which triggers a change feed in CosmosDB.
+
+Azure Functions configured to listen to the feed get a list of documents which are changed (add/edited/deleted). The Change feed listener convert them to events.
+
+The events are sent to Durable Entity orchestrator, which routes these event based on store identity to the virtual actor representing the store.
+
+The virtual actor processes the event and updates its states. It then commits the data to the Master Data Store container in Cosmos DB.
```

Comment:
You have been calling this "Master Data Container" in other places, should pick one name to use everywhere

Reply:
We need to make it consistent. Master Data Collection (MDC) is the business terminology. In the new CosmosDB terminology, we create "Containers" instead of "Partitioned Collections". So I was trying to keep up with that.

Reply:
"Container" is the generic term for any Cosmos DB API and "Collection" is specific to SQL/ Mongo as "Graph" is specific to Gremlin and "Table" is specific to Cassandra. In this case either Container or Collection would be correct, just needs to be consistent

Reply:
Thanks for the clarification @jcocchi